pynestml/codegeneration/resources_nest/point_neuron/SynapseHeader.h.jinja2

```diff
@@ -18,1332 +18,4 @@ GNU General Public License for more details.
-#endif
-  // purely dendritic delay
-  double dendritic_delay = get_delay();
-
-  // get spike history in relevant range (t_last_update, t_trig] from postsyn. neuron
-  std::deque< histentry__{{paired_neuron}} >::iterator start;
-  std::deque< histentry__{{paired_neuron}} >::iterator finish;
-  static_cast<{{paired_neuron}}*>(get_target(t))->get_history__( t_last_update_ - dendritic_delay, t_trig - dendritic_delay, &start, &finish );
-
-  // facilitation due to postsyn. spikes since last update
-  double t0 = t_last_update_;
-  // double minus_dt;
-  double timestep = 0;
-
-  while ( start != finish )
-  {
-{%- for vt_port in vt_ports %}
-{%- set vt_port = vt_ports[0] %}
-    process_{{vt_port}}_spikes_( vt_spikes, t0, start->t_ + dendritic_delay, cp );
-{%- endfor %}
-
-#ifdef DEBUG
-    std::cout << "\tprocessing post spike from " << t_last_update_ << " to " << start->t_ + dendritic_delay << std::endl;
-#endif
-
-    /**
-     * update synapse internal state from `t_last_update_` to `start->t_`
-    **/
-
-    update_internal_state_(t_last_update_,
-                           (start->t_ + dendritic_delay) - t_last_update_,
-                           cp);
-
-    const double _tr_t = start->t_;
-#ifdef DEBUG
-        std::cout << "\tFacilitating from c = " << S_.c << " (using trace = " << S_.pre_tr << ")";
-#endif
-{%- filter indent(6, True) %}
-{%- if post_ports is defined %}
-{%-     for post_port in spiking_post_ports %}
-    /**
-     *  NESTML generated onReceive code block for postsynaptic port "{{post_port}}" begins here!
-    **/
-
-{%-         set dynamics = synapse.get_on_receive_block(post_port) %}
-{%-         with ast = dynamics.get_block() %}
-{%-             include "directives/Block.jinja2" %}
-{%-         endwith %}
-{%-     endfor %}
-{%- endif %}
-{%- endfilter %}
-
-// #ifdef DEBUG
-//   std::cout << "\t--> new w = " << S_.w << std::endl;
-// #endif
-#ifdef DEBUG
-      std::cout << " to " << S_.c << std::endl;
-#endif
-    /**
-     * internal state has now been fully updated to `start->t_ + dendritic_delay`
-    **/
-
-    t0 = start->t_ + dendritic_delay;
-    // minus_dt = t_last_update_ - t0;
-    t_lastspike_ = start->t_ + dendritic_delay;
-    ++start;
-  }
-
-  /**
-    * update synapse internal state from `t_lastspike_` to `t_trig`
-  **/
-
-{%- for vt_port in vt_ports %}
-{%- set vt_port = vt_ports[0] %}
-  process_{{vt_port}}_spikes_( vt_spikes, t_lastspike_, t_trig, cp );
-{%- endfor %}
-
-#ifdef DEBUG
-    //std::cout << "{{synapseName}}::trigger_update_weight(): \tupdating from " << t_lastspike_ << " to " << t_trig + dendritic_delay << std::endl;
-#endif
-
- /* update_internal_state_(t_lastspike_,
-                         t_trig - t_lastspike_,
-                         cp);*/
-
-  vt_spikes_idx_ = 0;
-  t_lastspike_ = t_trig;
-}
-
-
-{%- endif %}
-
-} // namespace
-
-#endif /* #ifndef {{synapseName.upper()}}_H */
+{%- include "common/SynapseHeader.h.jinja2" %}
```

Comment:
If it doesn't include new changes, why is it being moved to a different place

Reply:
This is to support NEST2 compatibility, so the template in ``common`` can be included from the NEST3 templates directory, or from the NEST2 templates directory.